blog/2023-11-01-paralus-and-argocd-integration/index.md

```diff
@@ -0,0 +1,188 @@
+
+Key features of Argo CD include
+
+- **Declarative Application Configuration:** Applications are defined in Git repositories using YAML files, making it easy to version and manage configurations.
+- **Automated Synchronization:** Argo CD continuously monitors Git repositories for changes, ensuring that the cluster is always in the desired state.
+- **Role-Based Access Control (RBAC):** Argo CD allows you to define granular access control policies for users and teams, enhancing security.
+
+## Paralus for Zero Trust Kubernetes Access
+
+**Paralus** is a powerful tool designed to provide [zero-trust access to Kubernetes clusters](https://www.paralus.io/blog/zero-trust-security-kubernetes-access-using-paralus). Zero trust, as a security paradigm, assumes that threats can come from both inside and outside the network, and no entity should be implicitly trusted. Paralus integrates with Kubernetes to ensure that every interaction is authenticated and authorized, irrespective of the user's location.
+
+Key features of Paralus include
+
+- **Authentication:** Users must authenticate themselves before accessing the Kubernetes cluster. Paralus supports various authentication methods, such as username/password, single sign-on using openid connect (sso), and more.
+- **Role-Based Access Control:** Access to Kubernetes resources is determined by policies and roles, ensuring only authorized users can perform specific actions.
+- **Audit Trails:** Every action taken in the Kubernetes cluster is logged, providing transparency and accountability.
+
+## Use Case: Secure Access to Kubernetes with Argo CD and Paralus
+
+Let's dive into a practical use case to understand how Argo CD and Paralus work together to provide Zero Trust Kubernetes access.
+
+### Scenario
+
+Imagine a mid-sized tech company with multiple Kubernetes clusters deployed in various cloud providers and on-premises data centers. The company uses Argo CD for continuous deployment of applications and wants to strengthen the security posture of their clusters.
+
+### Using Argo CD to manage Paralus applications
+
+We can leverage Argo CD to deploy and manage Paralus core controller apps and relay agents across these multiple kubernetes clusters
+
+**Paralus Core:** Install and manage paralus app as an argocd helm application
+
+In this example we are going to deploy paralus core to the same cluster as argocd, you can choose a [different destination cluster](https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_cluster_add/). This would make paralus upgrades automatic on every release depending on your synchronization option and leverage the power for argocd.
+
+```
+argocd app create paralus-core
```

Comment:
Should we differentiate this command from Application yaml file present below?

Reply:
Even I was thinking back and forth, but finally decided to combine. Will seperate it though